include/boost/url/impl/const_string.hpp

```diff
@@ -92,8 +45,27 @@ const_string(
     std::size_t n,
     Allocator const& a,
     char*& dest)
-    : p_(construct(n, a, dest))
+    : refs_(nullptr),
+      alloc_(a)
 {
+    if(n == 0)
+    {
+        static_cast<string_view&>(*this) = { nullptr, n };
+        return;
+    }
+
+    using U = typename boost::type_with_alignment<
+        alignof(ref_counter_t)>::type;
+    auto constexpr S = sizeof(U);
+    using A = allocator_rebind_t<any_allocator<ref_counter_t>, U>;
+    A aa{alloc_};
+    refs_ = reinterpret_cast<ref_counter_t*>(
+        allocator_allocate(
+        aa, (sizeof(ref_counter_t) + n + S - 1) / S)
+        );
+    allocator_construct(alloc_, refs_, 1);
+    dest = reinterpret_cast<char*>(refs_ + 1);
+    static_cast<string_view&>(*this) = {dest, n};
```

Comment:
wouldn't this work?
```
*this = string_view( dest, n );
```
?

Reply:
```
error: No viable overloaded '='
```

If it went to the constructor, it would be (i) trying to construct again with std::allocator and (i) an infinite recursion. 
If it went to `operator=`, then it would be wrong because we can´t assign string_view to a constant string.